roadmap/implementers-guide/src/node/utility/availability-store.md

```diff
@@ -39,130 +33,138 @@ digraph {
+```rust
+struct CandidateMeta {
+  state: State,
+  data_available: bool,
+  chunks_stored: Bitfield,
+}
 
-For each head in the `activated` list:
-	- Note any new candidates backed in the block. Update pruning records for any stored `PoVBlock`s.
-	- Note any newly-included candidates backed in the block. Update pruning records for any stored availability chunks.
+enum State {
+  /// Candidate data was first observed at the given time but is not available in any block.
+  Unavailable(Timestamp),
+  /// The candidate was first observed at the given time and was included in the given list of unfinalized blocks, which may be
+  /// empty. The timestamp here is not used for pruning. Either one of these blocks will be finalized or the state will regress to
+  /// `State::Unavailable`, in which case the same timestamp will be reused.
+  Unfinalized(Timestamp, Vec<(BlockNumber, BlockHash)>),
+  /// Candidate data has appeared in a finalized block and did so at the given time.
+  Finalized(Timestamp)
+}
+```
 
-On `OverseerSignal::BlockFinalized(_)` events:
+We maintain the invariant that if a candidate has a meta entry, its available data exists on disk if `data_available` is true. All chunks mentioned in the meta entry are available.
 
-- Handle all pruning based on the newly-finalized block.
+Additionally, there is exactly one `prune_by_time` entry which holds the candidate hash unless the state is `Unfinalized`. There may be zero, one, or many "unfinalized" keys with the given candidate, and this will correspond to the `state` of the meta entry.
 
-On `QueryPoV` message:
+## Protocol
 
-- Return the PoV block, if any, for that candidate hash.
+Input: [`AvailabilityStoreMessage`][ASM]
 
-On `QueryChunk` message:
+Output:
+- [`RuntimeApiMessage`][RAM]
 
-- Determine if we have the chunk indicated by the parameters and return it and its inclusion proof via the response channel if so.
 
-On `StoreChunk` message:
+## Functionality
 
-- Store the chunk along with its inclusion proof under the candidate hash and validator index.
+For each head in the `activated` list:
+  - Note any new candidates backed in the block. Update the `CandidateMeta` for each. If the `CandidateMeta` does not exist, create it as `Unavailable` with the current timestamp. Register a `"prune_by_time"` entry based on the current timestamp + 1 hour.
+  - Note any new candidate included in the block. Update the `CandidateMeta` for each, performing a transition from `Unavailable` to `Unfinalized` if necessary. That includes removing the `"prune_by_time"` entry. Add the block hash and number to the state, if unfinalized. Add an `"unfinalized"` entry for the block and candidate.
+  - The `CandidateEvent` runtime API can be used for this purpose.
+  - TODO: load all ancestors of the head back to the finalized block so we don't miss anything if import notifications are missed. If a `StoreChunk` message is received for a candidate which has no entry, then we will prematurely lose the data.
```

Comment:
Within the guide, are we following the convention that TODOs require an associated issue?

Reply:
No, we haven't been yet.